### PR TITLE
Sj add punit

### PR DESCRIPTION
**Pull request by Shafaq-Juda**

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -67,15 +67,6 @@ def set_comment
       @comment = Comment.find(params[:id])
     end
 
-
-    def is_an_autorized_user
```
Comment (joannarodriguez134) [inline app/controllers/comments_controller.rb:71]: it works fine but authorized is misspelled

File: app/policies/photo_policy.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+class PhotoPolicy < ApplicationPolicy
+  
+  class Scope < Scope
+    # NOTE: Be explicit about which records you allow access to!
+    # def resolve
```
Comment (joannarodriguez134) [inline app/policies/photo_policy.rb:5]: Delete this comment

File: app/policies/photo_policy.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      @photo = photo
+     end
+
+   end
+  def index?
+    true
+  end
+
+    # Our policy is that a photo should only be seen by the owner or followers
+   #   of the owner, unless the owner is not private in which case anyone can see it
```
Comment (joannarodriguez134) [inline app/policies/photo_policy.rb:22]: Delete this comment

**Review by joannarodriguez134 — APPROVED** [review]

File: Gemfile
Comment (lsampson1974) [inline Gemfile:1]: Good

File: Gemfile.lock
Comment (lsampson1974) [inline Gemfile.lock:1]: This is good

File: app/controllers/application_controller.rb
Comment (lsampson1974) [inline app/controllers/application_controller.rb:1]: Good setup for authorization.

File: app/controllers/comments_controller.rb
Comment (lsampson1974) [inline app/controllers/comments_controller.rb:1]: Good for simplifying the authorization process.

File: app/controllers/follow_requests_controller.rb
Comment (lsampson1974) [inline app/controllers/follow_requests_controller.rb:1]: Nice

File: app/controllers/photos_controller.rb
Comment (lsampson1974) [inline app/controllers/photos_controller.rb:1]: Good.

File: app/controllers/users_controller.rb
Comment (lsampson1974) [inline app/controllers/users_controller.rb:1]: Good

File: app/policies/application_policy.rb
Comment (lsampson1974) [inline app/policies/application_policy.rb:1]: Good

File: app/policies/comment_policy.rb
Comment (lsampson1974) [inline app/policies/comment_policy.rb:1]: This is good.

File: app/policies/follow_request_policy.rb
Comment (lsampson1974) [inline app/policies/follow_request_policy.rb:1]: Good.

File: app/policies/photo_policy.rb
Comment (lsampson1974) [inline app/policies/photo_policy.rb:1]: Good

File: app/policies/user_policy.rb
Comment (lsampson1974) [inline app/policies/user_policy.rb:1]: Good.

File: app/views/comments/_comment.html.erb
Comment (lsampson1974) [inline app/views/comments/_comment.html.erb:1]: Nice

File: app/views/follow_requests/_form.html.erb
Comment (lsampson1974) [inline app/views/follow_requests/_form.html.erb:1]: Good

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -10,4 +12,10 @@ def set_user
...
       end
     end
-end
\ No newline at end of file
+
+    def must_be_owner_to_view
+      if current_user != @user
+        redirect_back fallback_location: root_url, alert: "You're not authorized"
+    end
+end
```
Comment (thierrychau) [inline app/controllers/users_controller.rb:20]: This should probably be in UserPolicy. Where do you call this?

File: app/policies/comment_policy.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    @user = user
+    @comment = comment
+  end 
+
+  def index?
+    true
+  end
+
+  def show?
+    user == comment.author
```
Comment (thierrychau) [inline app/policies/comment_policy.rb:15]: As is, a user can only see their own comments.

File: app/policies/photo_policy.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+   def show?
+     user == record.owner ||
```
Comment (thierrychau) [inline app/policies/photo_policy.rb:24]: why not use `author?` here?

**Review by thierrychau — COMMENTED** [review]
- Tested and looks good to me
- Called `verify_authorized` and `verify_policy_scoped`
- Each controller and action calls `authorize`
- Each controller and action have a policy

Good job, just a few minor comments below.

**Pull request merged**